src/backend/sev/snp/firmware/mod.rs

```diff
@@ -62,6 +62,12 @@ impl From<Identifier> for Vec<u8> {
     }
 }
 
+impl From<Vec<u8>> for Identifier {
+    fn from(vec: Vec<u8>) -> Self {
+        Identifier(vec)
```

Comment:
This does no validation that `vec` represents an actual valid identifier, but I am not sure we can actually assert that in any meaningful way.
We could check that it's at most 64 character long, since that seems like the maximum length of it according to the spec, but that seems to be about all we could do here and not sure if that's worth it, especially since the functionality is only used for testing.

Here's what the SEV spec has to say about it (SNP spec references SEV spec):
![image](https://user-images.githubusercontent.com/12877905/138312384-258f72e5-7991-478a-b930-63d56ebfa90f.png)

Reply:
The ID can be any length. We can't do any validation of it. The contents are opaque.